python/ray/data/_internal/util.py

```diff
@@ -397,3 +404,28 @@ def _split_list(arr: List[Any], num_splits: int) -> List[List[Any]]:
         arr[i * q + min(i, r) : (i + 1) * q + min(i + 1, r)] for i in range(num_splits)
     ]
     return splits
+
+
+def _best_batch_format(
```

Comment:
If this can be a util, why cannot it be the default impl for `DatasetIterator`?

Reply:
So `DatasetIterator` has no definition of `ds: "Dataset`. `PipelinedDatasetIterator` just uses `base_dataset_pipeline`. And `DatasetIteratorImpl` and `StreamSplitDatasetIterator` uses `_base_dataset`.